docs/upgrade.html

```diff
@@ -14,31 +14,73 @@
+
+<p><b>Note:</b> Bumping the protocol version and restarting can be done any time after the brokers were upgraded. It does not have to be immediately after.
+
+<ol>
+    <li>Upgrading a Kafka Streams Applications:
+        <ul>
+            <li>You need to recompile your code. Just swapping the jar file will not work and will break your appliation.</li>
+            <li>If you use a custom timestamp extractor, you will need to update this code, because the <code>TimestampExtractor</code> interface was changed.</li>
+        </ul>
+    </li>
 </ol>
 
 <h5><a id="upgrade_1020_notable" href="#upgrade_1020_notable">Notable changes in 0.10.2.0</a></h5>
 <ul>
+    <li>The Java clients (producer and consumer) have acquired the ability to communicate with older brokers. Version 0.10.2 clients
+        can talk to version 0.10.0 or newer brokers. Note that some features are not available or are limited when older brokers
+        are used. </li>
     <li>Several methods on the Java consumer may now throw <code>InterruptException</code> if the calling thread is interrupted.
         Please refer to the <code>KafkaConsumer</code> Javadoc for a more in-depth explanation of this change.</li>
     <li>Java consumer now shuts down gracefully. By default, the consumer waits up to 30 seconds to complete pending requests.
         A new close API with timeout has been added to <code>KafkaConsumer</code> to control the maximum wait time.</li>
     <li>Multiple regular expressions separated by commas can be passed to MirrorMaker with the new Java consumer via the --whitelist option. This
         makes the behaviour consistent with MirrorMaker when used the old Scala consumer.</li>
-    <li>Zookeeper dependency was removed from the Streams API. Now the Streams API uses Kafka protocol to manage internal topics instead of modifying Zookeeper directly. This eliminates the need for privilege[s] to access Zookeeper directly and "StreamsConfig.ZOOKEEPER_CONFIG" should not be set in the streams app anymore. If the Kafka cluster is secured, the Streams apps require to have proper security privilege[s] to create new topics.</li>
-    <li>Several new fields including "security.protocol", "connections.max.idle.ms", "retry.backoff.ms", "reconnect.backoff.ms" and "request.timeout.ms" were added to StreamsConfig class. User should pay attenntion to the default values and set these if needed. For more details please refer to <a id="streamsconfigs" href="#streamsconfigs">3.5 Kafka Streams Configs</a>.</li>
+    <li>The Zookeeper dependency was removed from the Streams API. The Streams API now uses the Kafka protocol to manage internal topics instead of
+        modifying Zookeeper directly. This eliminates the need for privileges to access Zookeeper directly and "StreamsConfig.ZOOKEEPER_CONFIG"
+        should not be set in the streams app any more. If the Kafka cluster is secured, Streams apps must have the required security privileges to create new topics.</li>
```

Comment:
typo: "streams app" -> "Streams app" (was already misspelled in the original text)